lib/galaxy/util/tool_shed/xml_util.py

```diff
@@ -91,7 +91,7 @@ def parse_xml(file_name, check_exists=True):
     if check_exists and not os.path.exists(file_name):
         return None, "File does not exist %s" % str(file_name)
 
-    with open(file_name, 'r') as fobj:
+    with io.open(file_name, 'r', encoding='utf-8') as fobj:
```

Comment:
ElementTree parses bytes internally, so the better fix is to not open `file_name` but pass it straight into `XmlET.parse(fobj, parser=XmlET.XMLParser(target=CommentedTreeBuilder()))`. This should handles non-utf8 document better.